tests/unit/test_store.py

```diff
@@ -25,6 +26,36 @@ async def test_store_defer(job_store, job_factory, connector):
     }
 
 
+async def test_store_defer_job_connector_exception(
+    mocker, job_store, job_factory, connector
+):
+    connector.execute_query_one = mocker.Mock(side_effect=exceptions.ConnectorException)
+
+    with pytest.raises(exceptions.ConnectorException):
+        await job_store.defer_job(job=job_factory(task_kwargs={"a": "b"}))
+
+
+async def test_store_defer_job_unique_violation_exception(
+    mocker, job_store, job_factory, connector
+):
+    class UniqueViolation(Exception):
+        pass
+
+    exc = exceptions.ConnectorException()
+    exc.__cause__ = UniqueViolation()
+    exc.__cause__.diag = mocker.Mock(
+        constraint_name="procrastinate_jobs_defer_lock_idx"
+    )
+
+    connector.execute_query_one = mocker.Mock(side_effect=exc)
+    mocker.patch.object(store.JobStore, "UniqueViolation", UniqueViolation)
+
+    with pytest.raises(exceptions.DeferLockTaken) as excinfo:
+        await job_store.defer_job(job=job_factory(task_kwargs={"a": "b"}))
+
+        assert excinfo.value.__cause__ is exc.__cause__
```

Comment:
@ewjoachim you want to like the mocks :-) If you don't I see no other ways than removing these unit tests entirely, and just relying on the corresponding integration tests.